wagtail/admin/views/workflows.py

```diff
@@ -393,24 +399,86 @@ def delete_action(self):
         self.object.deactivate(user=self.request.user)
 
 
-def usage(request, pk):
-    workflow = get_object_or_404(Workflow, id=pk)
+class WorkflowPageFilterSet(PageIndexView.filterset_class):
+    content_type = PageTypeFilter()
 
-    editable_pages = page_permission_policy.instances_user_has_permission_for(
-        request.user, "change"
-    )
-    pages = workflow.all_pages() & editable_pages
-    paginator = Paginator(pages, per_page=10)
-    pages = paginator.get_page(request.GET.get("p"))
 
-    return render(
-        request,
-        "wagtailadmin/workflows/usage.html",
-        {
-            "workflow": workflow,
-            "used_by": pages,
-        },
-    )
+class WorkflowUsageView(PageIndexView):
+    permission_policy = page_permission_policy
+    pk_url_kwarg = "pk"
+    index_url_name = "wagtailadmin_workflows:usage"
+    index_results_url_name = "wagtailadmin_workflows:usage_results"
+    header_icon = "tasks"
+    filterset_class = WorkflowPageFilterSet
+    page_title = _("Usage")
+    paginate_by = 20
+    # Avoid ambiguity of whether this view is working with the page or the workflow,
+    # and let the view code be more explicit about which model is being worked with.
+    model = None
+
+    @classproperty
+    def columns(cls):
+        columns = super().columns.copy()
+        columns.insert(
+            3,
+            Column(
+                "type",
+                label=_("Type"),
+                accessor="page_type_display_name",
+                sort_key="content_type",
+                width="12%",
+            ),
+        )
+        return columns
+
+    def dispatch(self, request, *args, **kwargs):
+        self.object = self.get_object()
+
+        # We set page_permission_policy as the main permission policy for this view,
+        # since this view is about listing pages that are using the workflow (and the
+        # base PageIndexView uses it to work with the queryset). However, we also want
+        # to ensure the user has permissions for the workflow itself.
+        if not workflow_permission_policy.user_has_any_permission_for_instance(
+            request.user, self.any_permission_required, self.object
+        ):
+            raise PermissionDenied
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_page_subtitle(self):
+        return self.object.name
+
+    def get_breadcrumbs_items(self):
+        title = self.get_page_title()
+        subtitle = self.get_page_subtitle()
+        return self.breadcrumbs_items + [
+            {
+                "url": reverse("wagtailadmin_workflows:index"),
```

Comment:
We should look into adding `self_url_name` or similar, to avoid ambiguity for "nested" listing views like this that may link to the index view of the object's model.